csrc/cutlass_extensions/epilogue/scaled_mm_epilogues_c3x.hpp

```diff
@@ -96,6 +107,14 @@ struct ScaledEpilogueBase {
                   std::is_same_v<Descriptor, RowLoad<T, true>>);
     return Arguments{data_ptr};
   }
+
+  template <typename Descriptor, typename T>
+  static auto args_from_tensor(const T* const* data_ptr, bool do_broadcast) {
+    using Arguments = typename Descriptor::Arguments;
+    static_assert(std::is_same_v<Descriptor, ColOrScalarLoadArray<T>> ||
+                  std::is_same_v<Descriptor, RowOrScalarLoadArray<T>>);
+    return Arguments{data_ptr, do_broadcast};
+  }
```

Comment:
Instead of requiring the caller to pass in a `do_broadcast` flag, can this function figure it out on its own (e.g. checking `tensor.numel() != 1` or similar)?

Reply:
This function takes as argument not the scales tensors, but the pointers to the start of each expert's scales, so I need to pass these booleans in addition to these pointers.

Reply:
What if you passed the tensors instead of just the pointers?